contrib/kubespray/script/openpai-generator.py

```diff
@@ -72,6 +84,61 @@ def generate_template_file(template_file_path, output_path, map_table):
     write_generated_file(output_path, generated_template)
 
 
+def get_kubernetes_node_info_from_API():
+    config.load_kube_config()
+    api_instance = client.CoreV1Api()
+
+    # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#list_node
+    pretty = 'true'
+    timeout_seconds = 56
+
+    ret = dict()
+    try:
+        api_response = api_instance.list_node(pretty=pretty, timeout_seconds=timeout_seconds)
+        for node in api_response.items:
+            ret[node.metadata.name] = {
+                "cpu-resource": int(parse_quantity(node.status.allocatable['cpu'])),
+                "mem-resource": int(parse_quantity(node.status.allocatable['memory']) / 1024 / 1024 ),
+                "gpu-resource": int(parse_quantity(node.status.allocatable['nvidia.com/gpu'])),
+            }
+    except ApiException as e:
+        logger.error("Exception when calling CoreV1Api->list_node: %s\n" % e)
+
+    print(ret)
+
+    return ret
+
+
+def hived_config_prepare(worker_dict, node_resource_dict):
+    hived_config = dict()
+    hived_config["nodelist"] = []
+
+    min_mem = 100000000 * 1024
+    min_gpu = 100000000
```

Comment:
this will overflow int32?

Reply:
just use 100000000 should be enough

---
In reply to: [366787443](https://github.com/microsoft/pai/pull/4142#discussion_r366787443) [](ancestors = 366787443)

Reply:
done